core/src/test/scala/sttp/tapir/ShadowedEndpointCheckerTest.scala

```diff
@@ -0,0 +1,116 @@
+package sttp.tapir
+
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+import sttp.tapir.util.ShadowedEndpointChecker
+
+class ShadowedEndpointCheckerTest extends AnyFlatSpecLike with Matchers {
+
+  it should "omit all segments which are not relevant for shadow check" in {
+    val e1 = endpoint.get.in(query[String]("key").and(header[String]("X-Account"))).in("x")
+    val e2 = endpoint.get.in("x")
+
+    val result = ShadowedEndpointChecker.apply(List(e1, e2))
+
+    val expectedResult = List(ShadowedEndpoint(e2, e1))
+    result shouldBe expectedResult
+  }
+
+  it should "should detect shadowed endpoints with path variables that contain special characters" in {
+    val e1 = endpoint.get.in("[x]")
+    val e2 = endpoint.get.in("x")
+    val e3 = endpoint.get.in("/..." / paths)
+    val e4 = endpoint.get.in("x/")
+    val e5 = endpoint.get.in("x")
+
+    val result = ShadowedEndpointChecker.apply(List(e1, e2, e3, e4, e5))
+
+    val expectedResult = List(ShadowedEndpoint(e5, e2))
+    result shouldBe expectedResult
+  }
+
+  it should "detect endpoint with wildcard shadowed by endpoint with fixed path" in {
+    val e1 = endpoint.get.in("x/y/z")
+    val e2 = endpoint.get.in("x" / "y" / "z")
+    val e3 = endpoint.get.in("x" / paths)
+
+    val result = ShadowedEndpointChecker.apply(List(e1, e2, e3))
+
+    result shouldBe Nil
+  }
+
+  it should "detect endpoint with path variable shadowed by endpoint with wildcard" in {
+    val e1 = endpoint.get.in("x" / paths)
+    val e2 = endpoint.get.in("x" / path[String].name("y_2") / "z")
+    val e3 = endpoint.get.in("x" / "y" / "x")
+
+    val result = ShadowedEndpointChecker.apply(List(e1, e2, e3))
+
+    val expectedResult = List(ShadowedEndpoint(e2, e1), ShadowedEndpoint(e3, e1))
+    result shouldBe expectedResult
+  }
+
+  it should "detect shadowed endpoints for endpoints with similar path structure but with different path vars names" in {
+    val e1 = endpoint.get.in("x" / "y")
+    val e2 = endpoint.get.in("x" / "y" / paths)
+    val e3 = endpoint.get.in("x" / path[String].name("y_2") / path[String].name("y_4") / "z1")
```

Comment:
is e3 shadowed by e2? e3 is `x/{}/{}/z1`, and e2 is `x/y/*`. There are requests which will be served by e3, e.g. `/x/a/b/z1`, and they won't be caught by e2